src/Diagnostics.DataProviders/KustoHeartBeatService/KustoHeartBeatService.cs

```diff
@@ -87,8 +87,17 @@ public async Task<string> GetClusterNameFromStamp(string stampName)
                             {
                                 var regionName = ((string)dr["Region"]).ToLower();
                                 var clusterName = (string)dr["ClusterName"];
-                                _configuration.RegionSpecificClusterNameCollection.TryAdd(regionName, clusterName + "follower");
-                                _configuration.FailoverClusterNameCollection.TryAdd(regionName, clusterName);
+                                if (_configuration.WawsPrimaryToDiagLeaderClusterMapping != null && _configuration.WawsPrimaryToDiagLeaderClusterMapping.ContainsKey(clusterName))
+                                {
+                                    string diagLeaderCluster = _configuration.WawsPrimaryToDiagLeaderClusterMapping[clusterName];
+                                    _configuration.RegionSpecificClusterNameCollection.TryAdd(regionName, diagLeaderCluster);
+                                    _configuration.FailoverClusterNameCollection.TryAdd(diagLeaderCluster, clusterName);
+                                }
+                                else 
+                                {
+                                    _configuration.RegionSpecificClusterNameCollection.TryAdd(regionName, clusterName + "follower");
+                                    _configuration.FailoverClusterNameCollection.TryAdd(clusterName + "follower", clusterName);
```

Comment:
Do we still need the follower cluster? The ultimate goal is to move away from followers and the failover cluster is the primary waws clusters as usual.

Reply:
I think until we completely remove the dependency on followers, we still need this. I will remove this when we move to diag leaders completely.